src/libponyc/expr/call.c

```diff
@@ -235,8 +280,16 @@ static bool check_arg_types(pass_opt_t* opt, ast_t* params, ast_t* positional,
       return false;
     }
 
-    ast_t* wp_type = consume_type(p_type, TK_NONE, false);
     errorframe_t info = NULL;
+    ast_t* wp_type;
+    if(check_auto_recover_newref(arg))
+    {
+      wp_type = unisolated(p_type);
```

Comment:
This isn't quite the right behavior, although it might be on the right track. You should be upgrading the argument/constructor capability instead of the parameter.


That way for instance, a `String val` argument can be fulfilled by auto-recovery of `String` (which is a `ref` constructor)

Reply:
I believe I've fixed this now.